centos2ol.sh

```diff
@@ -31,17 +38,68 @@ bad_packages=(centos-backgrounds centos-logos centos-release centos-release-cr d
               libreport-centos libreport-plugin-mantisbt libreport-plugin-rhtsupport python3-syspurpose \
               python-oauth sl-logos yum-rhn-plugin)
 
+function step_info
+{
+STEPNO++
+   echo
+   echo
+   echo "============================================================================"
+   echo "*** Step $STEP_NO, starting at LINE $1: $2"
+   echo "============================================================================"
+   if (( STARTNO > 0 && STEPNO < STARTNO )); then 
+      echo
+      echo "=== STEP SKIPPED ==="
+      echo
+   fi  
+      
+   echo $STEPNO > $BASE/laststep
+   if (( debug )); then 
+     answer='y'
+     echo "Continue ? (Enter or Y to continue, A to abort, integer -- skip to the given step)." 
+     read answer
+     if [[ $answer =~ [Aa] ]]; then
+        exit 1
+     fi
+     if  (( answer > 0 )) ; then
+         STARTNO=$answer;
+         echo Continuing with the step $answer 
+         return;
+     fi 
+   else
+     sleep 3
+   fi     
+}
+
+function info
+{
+   echo 
+   echo "INFO-$1: $2"
+   echo
+   sleep 3    
+}
+
+function abend
+{
+   echo
+   echo "============================================================================"
+   echo " !!! Fatal error at line $1: $2"
+   echo "For assistance, please open an issue via GitHub: ${github_url}."
+   echo "============================================================================"
+   echo
+       
+}
+
 usage() {
     echo "Usage: ${0##*/} [OPTIONS]"
     echo
     echo "OPTIONS"
     echo "-h"
     echo "        Display this help and exit"
+    echo "-d"
```

Comment:
I would probably call this "interactive" mode, which requires confirmation at each step.